### PR TITLE
257 replace mocks

### LINKED ISSUE #257: Replace mocks in `test_CatalogManager_load_invalid_model`
# Is your feature request related to a problem? Please describe.

The tests adde for more thorough catalog manager error handling in #255 relied on mocking exceptions raised from intake-dataframe-catalog.

I'm not really a fan of doing this as we're making assumptions about the error handling in that package, rather than the data, and so I would like to replace these mocks with a proper test of system response to  __data that should__ lead to the mocked exceptions being raised from `intake-dataframe-catalog`. 

This will take a bit of time to do, so I'm leaving it as an issue for now & will replace the mocks after release 1.0.0

### PR DESCRIPTION
**Pull request by charles-turner-1**
Closes #257, and makes some progress on #271.

Unfortunately I got mixed up and have removed a lot of the wrong mocks, but this PR currently replaces a lot of the (I think unnecessary) mocks in the `tests/test_cli.py` file, as well as replacing a lot of calls to the `os` module with `pathlib`.

Marked as draft because I still need to address the mocks in #257, but it seemed logical to bundle these all together.

@marc-white If you still remember the context of these tests, can you tell me if I'm violating any of the intentions with my changes?

### CODE REVIEW

File: tests/test_manager.py [hunk 1]
```diff
@@ -184,25 +184,53 @@ def test_CatalogManager_all(tmp_path, test_data):
...
+    )
+
+    with pytest.raises(CatalogManagerError) as excinfo:
+        cat.load(**load_args)
+
+    assert "Error adding source 'cmip5-al33' to the catalog" in str(excinfo.value)
+    assert (
+        "Expected iterable metadata columns: ['model', 'realm', 'frequency', 'variable']"
+        in str(excinfo.value.__cause__)
+    )
```
Comment (marc-white) [inline tests/test_manager.py:225]: This is currently throwing me an error, I think the `model` column is missing.

Comment (charles-turner-1) [reply tests/test_manager.py:225]: Weird - passes on my machine & CI.

Can you show me the error messaeg you're getting?

Comment (marc-white) [reply tests/test_manager.py:225]: From `pytest --cov-report=term-missing --cov=access_nri_intake`, I get:

```
FAILED tests/test_manager.py::test_CatalogManager_load_non_iterable - assert "Expected iterable metadata columns: ['model', 'realm', 'frequency', 'variable']" in "Cannot add entry with iterable metadata columns: ['realm', 'frequency', 'variable'] to dataframe catalog with iterable metadata columns: ['model', 'realm', 'frequency', 'variable'].  Please ensure that metadata ...
```

**Comment by charles-turner-1** [comment]
Extremely confused as to whats going on with test failures here - have moved back to draft

**Comment by marc-white** [comment]
> Extremely confused as to whats going on with test failures here - have moved back to draft

Maybe you've got a mixed checkout locally with some out-of-date test data?

**Comment by marc-white** [comment]
The test conversions themselves look fine to me.

**Comment by marc-white** [comment]
I still get a test failure on a local checkout copy:

```
FAILED tests/test_manager.py::test_CatalogManager_load_non_iterable - assert "Expected iterable metadata columns: ['model', 'realm', 'frequency', 'variable']" in "Cannot add entry with iterable metadata columns: ['realm', 'frequency', 'variable'] to dataframe catalog with iterable metadata columns: ['model', 'realm', 'frequency', 'variable'].  Please ensure that metadata entrie...
```

**Comment by charles-turner-1** [comment]
Weird - all passing on my machine & CI - will double check whats up.

File: tests/test_manager.py (hunk 1 above)
Comment (charles-turner-1) [reply tests/test_manager.py:225]: @marc-white Is it still this same error that you're getting?

**Comment by charles-turner-1** [comment]
> I still get a test failure on a local checkout copy:
> 
> ```
> FAILED tests/test_manager.py::test_CatalogManager_load_non_iterable - assert "Expected iterable metadata columns: ['model', 'realm', 'frequency', 'variable']" in "Cannot add entry with iterable metadata columns: ['realm', 'frequency', 'variable'] to dataframe catalog with iterable metadata columns: ['model', 'realm', 'frequency', 'variable'].  Please ensure that metadata entrie...
> ```

Okay, I'm confident I know the source of this - can you confirm the version of `intake-dataframe-catalog` you're using?  

This test is checking for the upstream error message. On reflection, this level of coupling is stupid, so I've replaced it with a try & a warning if it fails. Let me know if that solves the issue  for you.

**Review by marc-white — APPROVED** [review]
That latest update seemed to fix things. FYI, I'm on `intake-dataframe-catalog=0.2.4`.

**Comment by charles-turner-1** [comment]
Cool - I'm on 0.3 & I assume CI is too, so I guess that's the likely source of the error. Anyway, good catch.

**Pull request merged**